### PR TITLE
[Service Fabric] `az sf cluster node-type add`: Fix the unexpected error that 'StorageAccountsOperations' object has no attribute 'create'

### PR DESCRIPTION
**Pull request by a-santamaria**
**Related command**
az sf cluster node-type add

**Description**
Fix adding a new node type. It was failing because due to a storage account method name breaking change. Reported here https://github.com/Azure/azure-cli/issues/21697

also found on #22344 and #22315

**Testing Guide**

**History Notes**

[Service Fabric] `az sf cluster node-type add`: Fix the unexpected error that 'StorageAccountsOperations' object has no attribute 'create'

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Service Fabric

**Review by LukeSlev — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
Could you add some tests for these cases?

**Comment by navba-MSFT** [comment]
> Could you add some tests for these cases?

@a-santamaria for visibility

**Comment by navba-MSFT** [comment]
> Could you add some tests for these cases?

Thanks @zhoxing-ms .
@a-santamaria As requested in the above comment, Could you please add test cases for this PR so that it could be approved and then merged ?

**Comment by a-santamaria** [comment]
> an you provide the CLI command `az xxx` and the test command `azdev test xxx` and the test python file which not include in the pr ？

yeah I just pushed the tests changes: https://github.com/Azure/azure-cli/pull/22283/commits/fb8ab50d0c3b6564cf5d1e7e1d76641ca3e4c549#diff-bcbe83fb48caf55eb8ad02c2f23325d7b4c840f38414f68dbacf8cebb76990d6
 
the tests that fail with this are:
 azdev test test_primary_nt_add_remove_node --live
 azdev test test_add_secondary_node_type_add_remove_node --live

 

The command that works if I run outside of test is:
az sf cluster node add -g <groupName> -c <clusterName> --node-type <nodeTypeName> --number-of-nodes-to-add 2
 

it seems to me that the test connection is closing

**Comment by wangzelin007** [comment]
> > an you provide the CLI command `az xxx` and the test command `azdev test xxx` and the test python file which not include in the pr ？
> 
> yeah I just pushed the tests changes: [fb8ab50#diff-bcbe83fb48caf55eb8ad02c2f23325d7b4c840f38414f68dbacf8cebb76990d6](https://github.com/Azure/azure-cli/commit/fb8ab50d0c3b6564cf5d1e7e1d76641ca3e4c549#diff-bcbe83fb48caf55eb8ad02c2f23325d7b4c840f38414f68dbacf8cebb76990d6)
> 
> the tests that fail with this are: azdev test test_primary_nt_add_remove_node --live azdev test test_add_secondary_node_type_add_remove_node --live
> 
> The command that works if I run outside of test is: az sf cluster node add -g -c --node-type --number-of-nodes-to-add 2
> 
> it seems to me that the test connection is closing

Hi Alfredo,
This problem is very strange, I suggest that you can use @unittest.skip to skip tests first.
I have tried the following methods, but none of them work:
1. add time.sleep().
2. Increase connection_timeout and read_timeout.
3. add `--no-wait` to `az sf cluster node add/remove` command to support no wait.
4. add retry for self.cmd('az sf cluster node add')

It only fails when its run as a test...

**Comment by wangzelin007** [comment]
Hi [a-santamaria](https://github.com/a-santamaria)，
Please make sure to get this pr ready to merge before 2022/05/20.
Otherwise, this PR cannot catch up with the release of this sprint.
And the release time of the next sprint is 2022/07/05.

**Comment by a-santamaria** [comment]
Thanks @wangzelin007 I fixed the tests and skipped the ones that have the issue. @zhoxing-ms please take a look to merge

**Review by zhoxing-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/servicefabric/custom.py [hunk 1]
```diff
@@ -376,16 +379,19 @@ def _add_common_name(cluster, is_admin, certificate_common_name, certificate_iss
 
     patch_request = ClusterUpdateParameters(client_certificate_thumbprints=cluster.client_certificate_thumbprints,
                                             client_certificate_common_names=cluster.client_certificate_common_names)
-    return client.update(resource_group_name, cluster_name, patch_request)
+    update_cluster_poll = client.begin_update(resource_group_name, cluster_name, patch_request)
+    return LongRunningOperation(cli_ctx)(update_cluster_poll)
```
[the remark this answers is not in the document]
Comment (a-santamaria) [reply src/azure-cli/azure/cli/command_modules/servicefabric/custom.py:383]: ok thanks, I will work on this recommendation on a separate pr

**Pull request merged**